spec/shared_contexts.rb

```diff
@@ -0,0 +1,72 @@
+RSpec.shared_context 'sqlite3 database setup' do |type|
+	module Daru::IO::Rspec
+	  class Account < ActiveRecord::Base
+	    self.table_name = 'accounts'
+	  end
+	end
+
+  before do
+    FileUtils.rm(db_name) if File.file?(db_name)
+    SQLite3::Database.new(db_name).tap do |db|
+      db.execute "create table accounts(id integer, name varchar, age integer, primary key(id))"
+      db.execute "insert into accounts values(1, 'Homer', 20)"
+      db.execute "insert into accounts values(2, 'Marge', 30)"
+    end
+    Daru::IO::Rspec::Account.establish_connection "sqlite3:#{db_name}"
+  end
+
+  let(:db_name)  { 'daru_test' }
+  let(:relation) { Daru::IO::Rspec::Account.all }
+
+  subject {
+    if defined? fields
+      Daru::IO::Importers::Activerecord.load relation, *fields
+    else
+      Daru::IO::Importers::Activerecord.load relation
+    end      
+
+  }
+
+  after { FileUtils.rm(db_name) }
+end
+
+RSpec.shared_context 'csv exporter setup' do
+  before { Daru::IO::Exporters::CSV.write df, tempfile.path, opts }
+
+  let(:tempfile) { Tempfile.new('data.csv') }
+  let(:opts)     { {} }
+  let(:df)       { 
+    Daru::DataFrame.new({
+      'a' => [1,2,3,4,5],
+      'b' => [11,22,33,44,55],
+      'c' => ['a', 'g', 4, 5,'addadf'],
+      'd' => [nil, 23, 4,'a','ff']})
+  }
+
+  subject { File.open(tempfile.path, &:readline).chomp.split(',', -1) }
+end
+
+RSpec.shared_context 'excel exporter setup' do
```

Comment:
Why this (and similar) "shared contexts"? They are used exactly at 1 point in code, and I don't believe making them shared brings any clarity.

I'd rather prefer to see some generic "exporter shared context" (creating dataframe for export, which is pretty similar everywhere).